odml/doc.py

```diff
@@ -14,23 +16,33 @@ class Document(base._baseobj):
 class BaseDocument(base.sectionable, Document):
     """
     A represenation of an odML document in memory.
-
     Its odml attributes are: *author*, *date*, *version* and *repository*.
-
     A Document behaves very much like a section, except that it cannot hold
     properties.
     """
 
     _format = format.Document
 
-    def __init__(self, author=None, date=None, version=None, repository=None):
+    def __init__(self, author=None, date=None, version=None, repository=None, id=None):
         super(BaseDocument, self).__init__()
+        self._id = str(uuid.uuid4())
         self._author = author
         self._date = date  # date must be a datetime
         self._version = version
         self._repository = repository
 
     @property
+    def id(self):
+        """
+        The uuid for the document.
+        """
+        return self._id
+
+    @id.setter
+    def id(self, new_value):
+        self._id = new_value
```

Comment:
I think you are right, we do not need the setter so far. While developing I used to think we would use it, but now it is clear that it is redundant.